### PR TITLE
Bug 1744532: Adds machineCIDR noProxy support for all platforms

### PR DESCRIPTION
**Pull request by danehans**
Previously, machineCIDR was only being added to the default noProxy list for clusters installed in cloud provider environments. According to https://bugzilla.redhat.com/show_bug.cgi?id=1744532, machineCIDR is used for vSphere and bare metal installations too. This PR adds machineCIDR to the default noProxy list for all installation types, provides consistency with PR https://github.com/openshift/installer/pull/2286

### CODE REVIEW

File: pkg/util/proxyconfig/merge.go [hunk 1]
```diff
@@ -20,11 +20,30 @@ import (
...
+	type installConfig struct {
+		ControlPlane struct {
+			Replicas string `json:"replicas"`
+		} `json:"controlPlane"`
+		Networking struct {
+			MachineCIDR string `json:"machineCIDR"`
+		} `json:"networking"`
+	}
+	var ic installConfig
+	data, ok := cluster.Data["install-config"]
```
Comment (squeed) [inline pkg/util/proxyconfig/merge.go]: Do all clusters have an install-config? Including UPI clusters?

Comment (danehans) [reply pkg/util/proxyconfig/merge.go]: @squeed yes, all install types include an install-config with machineCIDR. See the [user docs ](https://github.com/openshift/installer/blob/master/docs/user) as a reference. A similar installer [PR](https://github.com/openshift/installer/pull/2286) was just merged.

**Pull request merged**